meta-lmp-base/recipes-extended/ostree/ostree/0001-ostree-pull-set-request-timeout.patch

```diff
@@ -0,0 +1,21 @@
+diff --git a/src/libostree/ostree-fetcher-curl.c b/src/libostree/ostree-fetcher-curl.c
+index 0ce3ff00..d50d62e7 100644
+--- a/src/libostree/ostree-fetcher-curl.c
++++ b/src/libostree/ostree-fetcher-curl.c
+@@ -817,6 +817,16 @@ initiate_next_curl_request (FetcherRequest *req,
+   curl_easy_setopt (req->easy, CURLOPT_WRITEDATA, task);
+   curl_easy_setopt (req->easy, CURLOPT_PROGRESSDATA, task);
+ 
++
++  /* set a request timeout, make sure it's not 0, otherwise an overall ostree pull session might hang */
++  long curl_timeout = 0L;
++  const char* curl_timeout_str = g_getenv ("OSTREE_CURL_TIMEOUT");
```

Comment:
Do we want to allow this value to be customized via an environment variable or should we just try to force a sane default?

Reply:
IMO, it might be useful for debugging as well as customization/tuning for a specific customer, we might wanna have a different value for devices with different connectivity types, e.g. LTE vs 1Gb Ethernet.

Reply:
Sure, seems reasonable.